### DISCUSSION TITLE
LDAP auth api

### DISCUSSION THREAD

**Discussion by kosie99** [directus/directus#10217] [Archive]
I am trying to get LDAP auth to work and reading the docs does not help me much.
I want to use the sAMAccountname to use as the username.
I am using the docker image and has set up all the AUTH_LDAP_... env variables.

What is the auth api url I should use?

Also, is a local user created upon successful ldap authentication?
If I am missing some documentation that describes it all, please point me to it :)

Regards,
Kosie

**Comment by aidenfoxx** [comment]
LDAP is documented here https://docs.directus.io/self-hosted/config-options.html#ldap-ldap

The variable you want is: `AUTH_<PROVIDER>_USER_ATTRIBUTE="sAMAccountname"`.

The LDAP login will create a new users upon successful authentication.

**Comment by kosie99** [comment]
Thanks Aiden,

I have my env vars set up as per the doc you provided, but I am still
unclear on how to contruct the auth url to make use of ldap.

I assume it is not just http://server/auth/login anymore?

Regards,
Kosie

**Reply by aidenfoxx** [reply]
LDAP runs under its own protocol. Usually it's "ldap://mydomain.ad" or "ldaps://mydomain.ad". Your sysadmin will probably be able to help you figure out what address you need.

EDIT: Unless you mean "how do I login?", in which case there should be an LDAP login option under a dropdown on the directus login page.

**Comment by kosie99** [comment]
I know how to connect to ldap, but I assume that I need to make an api call
to directus, which will do the actual auth on ldap? So, I need to
understand what the api is that I should use to connect to Directus in
order for it to use ldap for auth instead of its local database.

Something like /auth/login/ldap perhaps?

Unless it does not work that way at all!

Kosie

**Reply by aidenfoxx** [reply]
Please reply in the thread. Just keeps the discussion easier to follow for others.

Like I said, you should be able to log in with LDAP via the directus login page. If you want to call the API directly you can do that via `/auth/login/<LDAP_PROVIDER_NAME>` with a POST request containing `{ identifier, password }`.

**Reply by kosie99** [reply]
Thanks Aiden, yes I want to use the API from a react frontend. I will try your suggestion there, thanks again!

Kosie

**Reply by quentinglorieux** [reply]
Hi there, 
This call jsut  access_token, refresh_token and expires but no user info. 
Is it normal ?
How to get user info while login with LDAP ?
